code/modules/clothing/suits/armor.dm

```diff
@@ -248,25 +255,73 @@
 		owner.visible_message("<span class='danger'>[owner] is hit by [attack_text] in the chest!</span>") //We pretend to be hit, since blocking it would stop the message otherwise
 		spawn(40)
 			owner.alpha = initial(owner.alpha)
+		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
 		return 1
 
 /obj/item/clothing/suit/armor/reactive/tesla
 	name = "reactive tesla armor"
+	desc = "An experimental suit of armor with sensitive detectors hooked up to a huge capacitor grid, with emitters strutting out of it. Zap."
 
 /obj/item/clothing/suit/armor/reactive/tesla/hit_reaction(mob/living/carbon/human/owner, attack_text)
 	if(!active)
 		return 0
 	if(prob(hit_reaction_chance))
+		if(world.time < reactivearmor_cooldown)
+			var/datum/effect_system/spark_spread/sparks = new /datum/effect_system/spark_spread
+			sparks.set_up(1, 1, src)
+			sparks.start()
+			owner.visible_message("<span class='danger'>The tesla capacitors on [owner]'s reactive telsa armor are still recharging! The armor merely emits some sparks.</spawn>")
+			return
 		owner.visible_message("<span class='danger'>The [src] blocks the [attack_text], sending out arcs of lightning!</span>")
 		for(var/mob/living/M in view(6, owner))
 			if(M == owner)
 				continue
 			owner.Beam(M,icon_state="lightning[rand(1, 12)]",icon='icons/effects/effects.dmi',time=5)
 			M.adjustFireLoss(25)
 			playsound(M, 'sound/machines/defib_zap.ogg', 50, 1, -1)
+		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
 		return 1
-//All of the armor below is mostly unused
 
+/obj/item/clothing/suit/armor/reactive/table
+	name = "reactive table armor"
+	desc = "If you can't beat the memes, embrace them."
+	var/tele_range = 10
+
+/obj/item/clothing/suit/armor/reactive/table/hit_reaction(mob/living/carbon/human/owner, attack_text)
+	if(!active)
+		return 0
+	if(prob(hit_reaction_chance))
+		var/mob/living/carbon/human/H = owner
+		if(world.time < reactivearmor_cooldown)
+			owner.visible_message("<span class='danger'>The reactive table armor's fabricators are still on cooldown!</span>")
+			return
+		owner.visible_message("<span class='danger'>The reactive teleport system flings [H] clear of [attack_text] and slams them into a fabricated table!</span>")
+		owner.visible_message("<font color='red' size='3'>[H] GOES ON THE TABLE!!!</font>")
+		owner.Weaken(2)
+		var/list/turfs = new/list()
+		for(var/turf/T in orange(tele_range, H))
+			if(T.density)
+				continue
+			if(T.x>world.maxx-tele_range || T.x<tele_range)
+				continue
+			if(T.y>world.maxy-tele_range || T.y<tele_range)
+				continue
+			turfs += T
+		if(!turfs.len)
+			turfs += pick(/turf in orange(tele_range, H))
+		var/turf/picked = pick(turfs)
+		if(!isturf(picked))
+			return
+		H.forceMove(picked)
+		new /obj/structure/table(get_turf(owner))
```

Comment:
please make this use a typepath var in the object, so it can be adminedited to slam people onto glass tables or bartables instead